mycore-base/src/main/java/org/mycore/access/MCRAccessCacheManager.java

```diff
@@ -96,4 +105,28 @@ public void removePermission(String id, String permission) {
         permissionCache.remove(handle);
     }
 
+    public void removePermission(String... ids) {
+        MCRCache<MCRPermissionHandle, Boolean> permissionCache = accessCache.get();
+        removePermissionFromCache(permissionCache, ids);
+    }
+
+    private void removePermissionFromCache(MCRCache<MCRPermissionHandle, Boolean> permissionCache, String... ids) {
+        Set<String> idSet = Stream.of(ids).collect(Collectors.toSet());
+        final List<MCRPermissionHandle> handlesToRemove = permissionCache.keys()
+            .stream()
+            .filter(hdl-> hdl.getId()!=null)
+            .filter(hdl -> idSet.contains(hdl.getId()))
+            .collect(Collectors.toList());
+        handlesToRemove.forEach(permissionCache::remove);
```

Comment:
Why do you not call forEach on the stream directly? There is no need to save the intermediate result als List.
```
 permissionCache.keys()
            .stream()
            .filter(hdl-> hdl.getId()!=null)
            .filter(hdl -> idSet.contains(hdl.getId()))
            .forEach(permissionCache::remove);
```

Reply:
This would cause a java.util.ConcurrentModificationException, because we are iterating over the same collection where the items are removed.

Reply:
O.K., your're right